blockstack_client/backend/drivers/ipfs.py

```diff
@@ -0,0 +1,410 @@
+    """
+    Get a chunk of data from IPFS.
+    
+    Return the data on success
+    Return None on error, and log an exception.
+    """
+    
+    api = ipfsapi.connect(IPFS_SERVER, IPFS_PORT)
+
+    data = None
+    begin = None
+    end = None
+    size = None
+    try:
+        begin = time.time()
+        if is_mutable:
+          r = api.name_resolve( chunk_path )
+          url = r['Path']
+        else:
+          url = chunk_path
+        compressed_data = api.cat(url)
+        end = time.time()
+        size = len(compressed_data)
+
+        try:
+            data = decompress_chunk( compressed_data )
+        except:
+            data = compressed_data
+        
+    except Exception, e:
+        log.error("Failed to read '%s'" % chunk_path)
+        log.exception(e)
+        
+    if os.environ.get("BLOCKSTACK_TEST") == "1":
+        log.debug("[BENCHMARK] s3.read_chunk %s: %s" % (size, end - begin))
+    return data
+
+#-------------------------
+def delete_chunk( chunk_path, is_mutable ):
+    """
+    Delete a chunk of data from IPFS.
+    
+    Return True on success 
+    Return False on error.
+    """
+    
+    api = ipfsapi.connect(IPFS_SERVER, IPFS_PORT)
+
+    rc = True
+
+    if is_mutable:
+      r = api.name_resolve( chunk_path )
+      url = r['Path']
+    else:
+      url = chunk_path
+
+    try:
+        api.pin_rm( url )
+    except Exception, e:
+        log.error("Failed to delete '%s'" % chunk_path)
+        log.exception(e)
+        rc = False
+
+    return rc
+
+
+def storage_init(conf, **kwargs):
+   """
+   Initialize IPFS storage driver
+   """
+   global IPFS_SERVER, IPFS_PORT, IPFS_COMPRESS
+
+   # path to the CLI's configuration file (where you can stash driver-specific configuration)
+   config_path = conf['path']
+   if os.path.exists( config_path ):
+
+       parser = SafeConfigParser()
+        
+       try:
+           parser.read(config_path)
+       except Exception, e:
+           log.exception(e)
+           return False
+
+       if parser.has_section('ipfs'):
+            
+            if parser.has_option('ipfs', 'server'):
+                IPFS_SERVER = parser.get('ipfs', 'server')
+                
+            if parser.has_option('ipfs', 'port'):
+                IPFS_PORT = parser.get('ipfs', 'port')
+            
+            if parser.has_option('ipfs', 'compress'):
+                AWS_COMPRESS = (parser.get('ipfs', 'compress', 'false').lower() in ['true', '1'])
+
+   return True 
+
+
+def handles_url( url ):
+    """
+    Does this storage driver handle this kind of URL?
+
+    It is okay if other drivers say that they can handle it.
+    This is used by the storage system to quickly filter out
+    drivers that don't handle this type of URL.
+
+    A common strategy is simply to check if the scheme
+    matches what your driver does.  Another common strategy
+    is to check if the URL matches a particular regex.
+    """
+    pattern = re.compile("Qm[1-9A-HJ-NP-Za-km-z]{44}")
```

Comment:
this regex should be changed to: check if it starts with`/ipfs/` or `/ipns/`, otherwise, check if it starts with a valid CID: https://github.com/ipld/cid  The Qm hashes are only the old style sha2-256 hashes, we have support for a few different other things now.